docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx

```diff
@@ -364,6 +365,226 @@ export const ErrorAfter2s = () => {
+    <ErrorBoundary fallback={({ error }) => <>{error.message} of Parent</>}>
+      <ErrorBoundary
+        shouldCatch={CustomError}
+        fallback={({ error }) => <>{error.message} of Child</>}
+      >
+        <CustomErrorAfter2s />
+      </ErrorBoundary>
+    </ErrorBoundary>
+  )
+}
+```
+
+  </Tabs.Tab>
+  <Tabs.Tab>
+  
+```tsx {10}
+import { ErrorBoundary } from '@suspensive/react'
+import { createElement } from 'react'
+import { CustomError } from './CustomError'
+import { CustomErrorAfter2s } from './CustomErrorAfter2s'
+
+export const Example = () => {
+  return (
+    <ErrorBoundary fallback={({ error }) => <>{error.message} of Parent</>}>
+      <ErrorBoundary
+        shouldCatch={(error) => error instanceof CustomError}
+        fallback={({ error }) => <>{error.message} of Child</>}
+      >
+        <CustomErrorAfter2s />
+      </ErrorBoundary>
+    </ErrorBoundary>
+  )
+}
+```
+  </Tabs.Tab>
+</Tabs>
+
+You can also apply multiple criteria through array.
+
+```tsx {10}
+import { ErrorBoundary } from '@suspensive/react'
+import { createElement } from 'react'
+import { CustomError } from './CustomError'
+import { CustomErrorAfter2s } from './CustomErrorAfter2s'
+
+export const Example = () => {
+return (
+  <ErrorBoundary fallback={({ error }) => <>{error.message} of Parent</>}>
+    <ErrorBoundary
+      shouldCatch={[false, CustomError, (error) => error instanceof CustomError]}
+      fallback={({ error }) => <>{error.message} of Child</>}
+    >
+      <CustomErrorAfter2s />
+    </ErrorBoundary>
+  </ErrorBoundary>
+  )
+}
+```
+
+<Sandpack>
+
+```tsx Example.tsx active
+import { ErrorBoundary } from '@suspensive/react'
+import { createElement } from 'react'
+import { CustomError } from './CustomError'
+import { CustomErrorAfter2s } from './CustomErrorAfter2s'
+
+export const Example = () => {
+  return (
+    <ErrorBoundary fallback={({ error }) => <>{error.message} of Parent</>}>
+      <ErrorBoundary
+        shouldCatch={CustomError}
+        fallback={({ error }) => <>{error.message} of Child</>}
+      >
+        <CustomErrorAfter2s />
+      </ErrorBoundary>
+    </ErrorBoundary>
+  )
+}
+```
+
+```tsx CustomError.ts
+
+export class CustomError extends Error {
+  constructor(...args: ConstructorParameters<ErrorConstructor>) {
+    super(...args)
+    console.error(...args)
+  }
+}
+```
+
+```tsx CustomErrorAfter2s.tsx
+import { useState, useEffect } from "react";
+import { CustomError } from "./CustomError";
+
+export const CustomErrorAfter2s = () => {
+  const [asyncState, setAsyncState] = useState<
+    { isError: true; error: CustomError } | { isError: false; error: null }
+  >({
+    isError: false,
+    error: null
+  });
+
+  useEffect(() => {
+    setTimeout(() => {
+      setAsyncState({
+        isError: true,
+        error: () => new CustomError("unauthorized")
+      });
+    }, 2000);
+  }, []);
+
+  if (asyncState.isError) {
+    throw asyncState.error();
+  }
+
+  return <>No error</>;
+};
+```
+
```

Comment:
In my opinion, no depth and one file could make doc easier to understand .

```suggestion
```tsx Example.tsx active
import { useState, useEffect } from "react";
import { ErrorBoundary } from '@suspensive/react'
import { createElement } from 'react'

export const Example = () => {
  return (
    <ErrorBoundary fallback={({ error }) => <>{error.message} of Parent</>}>
      <ErrorBoundary
        shouldCatch={CustomError}
        fallback={({ error }) => <>{error.message} of Child</>}
      >
        <CustomErrorAfter2s />
      </ErrorBoundary>
    </ErrorBoundary>
  )
}

class CustomError extends Error {
  constructor(...args: ConstructorParameters<ErrorConstructor>) {
    super(...args)
    console.error(...args)
  }
}

export const CustomErrorAfter2s = () => {
  const [asyncState, setAsyncState] = useState<
    { isError: true; error: CustomError } | { isError: false; error: null }
  >({
    isError: false,
    error: null
  });

  useEffect(() => {
    setTimeout(() => {
      setAsyncState({
        isError: true,
        error: () => new CustomError("unauthorized")
      });
    }, 2000);
  }, []);

  if (asyncState.isError) {
    throw asyncState.error();
  }

  return <>No error</>;
};
```

```

Reply:
Oh, Okay. I fixed it. 🧐